### PR TITLE
Upgrade to Calcite 1.15.0

### LINKED ISSUE #4635: SQL: NPE on filtered aggregation with "NOT IN"
This query throws an NPE during planning.

```sql
SELECT
  COUNT(*) filter(WHERE dim1 NOT IN ('1'))
  COUNT(dim2) filter(WHERE dim1 NOT IN ('1'))
FROM druid.foo
```

Seems to be a Calcite bug, https://issues.apache.org/jira/browse/CALCITE-1910.

The stack trace:

```
java.lang.NullPointerException
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:212)
        at org.apache.calcite.sql2rel.SqlToRelConverter$Blackboard.convertExpression(SqlToRelConverter.java:4426)
        at org.apache.calcite.sql2rel.SqlToRelConverter$AggConverter.translateAgg(SqlToRelConverter.java:4888)
        at org.apache.calcite.sql2rel.SqlToRelConverter$AggConverter.visit(SqlToRelConverter.java:4819)
        at org.apache.calcite.sql2rel.SqlToRelConverter$AggConverter.visit(SqlToRelConverter.java:4663)
        at org.apache.calcite.sql.SqlCall.accept(SqlCall.java:137)
        at org.apache.calcite.sql2rel.SqlToRelConverter$AggConverter.visit(SqlToRelConverter.java:4795)
        at org.apache.calcite.sql2rel.SqlToRelConverter$AggConverter.visit(SqlToRelConverter.java:4663)
        at org.apache.calcite.sql.SqlNodeList.accept(SqlNodeList.java:153)
        at org.apache.calcite.sql2rel.SqlToRelConverter.createAggImpl(SqlToRelConverter.java:2715)
        at org.apache.calcite.sql2rel.SqlToRelConverter.convertAgg(SqlToRelConverter.java:2656)
        at org.apache.calcite.sql2rel.SqlToRelConverter.convertSelectImpl(SqlToRelConverter.java:658)
        at org.apache.calcite.sql2rel.SqlToRelConverter.convertSelect(SqlToRelConverter.java:620)
        at org.apache.calcite.sql2rel.SqlToRelConverter.convertQueryRecursive(SqlToRelConverter.java:3066)
        at org.apache.calcite.sql2rel.SqlToRelConverter.convertQuery(SqlToRelConverter.java:556)
```

### PR DESCRIPTION
**Pull request by jon-wei**
This PR upgrades Calcite to 1.15.0, which fixes #4635 and #4945, with corresponding unit tests.

Calcite 1.15.0 changed the way it handles `EXTRACT(EPOCH ...)`, so this patch updates Druid accordingly.

The PR also changes Druid's EPOCH units from milliseconds to seconds, which is more standard, e.g.:
https://www.postgresql.org/docs/9.1/static/functions-datetime.html
https://www.ibm.com/support/knowledgecenter/en/SSULQD_7.2.1/com.ibm.nz.dbu.doc/r_dbuser_functions_expressions_extract_datetime.html
  
The PR also re-enables/updates unit tests that were affected by https://issues.apache.org/jira/browse/CALCITE-1601 (fixed in Calcite 1.12.0, but the Druid tests were not enabled)

### CODE REVIEW

File: sql/src/test/java/io/druid/sql/calcite/CalciteQueryTest.java [hunk 1]
```diff
@@ -4928,9 +4994,15 @@ public void testFilterOnTimeExtract() throws Exception
         ImmutableList.of(
             Druids.newTimeseriesQueryBuilder()
                   .dataSource(CalciteTests.DATASOURCE1)
-                  .intervals(QSS(Intervals.of("2000/P1M")))
+                  .intervals(QSS(Intervals.of("-146136543-09-08T08:23:32.096Z/146140482-04-24T15:36:27.903Z")))
```
Comment (gianm) [inline sql/src/test/java/io/druid/sql/calcite/CalciteQueryTest.java]: This would be better as `.intervals(QSS(Filtration.eternity()))`

Comment (jon-wei) [reply sql/src/test/java/io/druid/sql/calcite/CalciteQueryTest.java]: thanks, changed to Filtration.eternity()

**Comment by himanshug** [comment]
@jon-wei it appears that calcite-core is now bringing memory-0.9.0.jar which is conflicing with memory-0.10.3.jar brought by datasketches module.

**Comment by jon-wei** [comment]
@himanshug I opened a PR that excludes sketches-core from calcite here: https://github.com/druid-io/druid/pull/5223

**Pull request merged**